rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -84,7 +84,10 @@ module Impl {
     // an enum constant (e.g. `None`). This excludes static and constant variables (UPPERCASE),
     // which we don't appear to recognize yet anyway. This also assumes programmers follow the
     // naming guidelines, which they generally do, but they're not enforced.
-    not name.charAt(0).isUppercase()
+    not name.charAt(0).isUppercase() and
+    // exclude parameters from functions without a bodies as these are trait method declarations
+    // without implementations
+    not exists(Function f | not f.hasBody() and f.getParamList().getAParam().getPat() = p)
```

Comment:
Note that this just compares the `IdentPat` directly with the pattern from the parameter, whereas the `getParameter` predicate does some more complicated work to get a parameter. However, patterns aren't allowed in parameters for functions without bodies, so I think this should be sufficient in this case.